### PR TITLE
scorcard api call integration nexB#2

### PR DESCRIPTION
**Pull request by 404-geek**
This commit contains some basic changes to the repo and API and cli calls to OSSF Apis

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,62 +1,3 @@
-A Simple Python Project Skeleton
-================================
-This repo attempts to standardize the structure of the Python-based project's
-repositories using modern Python packaging and configuration techniques.
-Using this `blog post`_ as inspiration, this repository serves as the base for
-all new Python projects and is mergeable in existing repositories as well.
+# ossf-scorecard
```
Comment (AyanSinhaMahapatra) [inline README.rst:1]: This is a `.rst` file, you're using markdown syntax. See https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections.

A bit of additional documentation and links to projects would be nice here.

File: src/ossf_scorecard/scorecard_cli.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  `openssf.scorecardcron.scorecard-v2`,
+  UNNEST(checks) AS check
+WHERE
+  repo.name = "github.com/pallets/flask"
+
+  LIMIT 100
+"""
+# AND scorecard.version = "v4.13.1-242-g153e06d9"
+
+query_job = client.query(query)
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/scorecard_cli.py:23]: Will this work? Don't we need additional settings to access bigquery?

File: src/ossf_scorecard/scorecard.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import requests
+
+
+def get_scorecard(platform, org, repo):
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/scorecard.py:4]: We also need to check if these inputs are valid.

File: src/ossf_scorecard/scorecard.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    url = f"https://api.securityscorecards.dev/projects/{platform}/{org}/{repo}"
+    response = requests.get(url)
+
+    if response.status_code == 200:
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/scorecard.py:8]: Can there be timeouts or other status codes, is there any api rate limiting here?

File: src/ossf_scorecard/scorecard.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        return response.json()
+    else:
+        response.raise_for_status()
+
+
+def print_scorecard(platform, org, repo):
+    try:
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/scorecard.py:15]: Let's not print, or have this function. If we want to print something we can use the log, but seems unnecessary here. We only have to store the data correctly in the models at https://github.com/nexB/ScoreCode/pull/4

File: src/ossf_scorecard/scorecard_cli.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  repo.name AS repo_name,
+  scorecard,
+  check.name AS check_name,
+  check.score AS check_score,
+  check.reason AS check_reason
+FROM
+  `openssf.scorecardcron.scorecard-v2`,
+  UNNEST(checks) AS check
+WHERE
+  repo.name = "github.com/pallets/flask"
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/scorecard_cli.py:17]: This is hardcoded, and not usable.

File: src/ossf_scorecard/scorecard_cli.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+  LIMIT 100
+"""
+# AND scorecard.version = "v4.13.1-242-g153e06d9"
+
+query_job = client.query(query)
+
+results = query_job.result()
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/scorecard_cli.py:25]: Can this actually be tested? Do we need to mock api calls?

File: src/ossf_scorecard/__init__.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#
```
Comment (AyanSinhaMahapatra) [inline src/ossf_scorecard/__init__.py:1]: We need to have this notice at every file.

File: tests/test_scorecard.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+class TestScorecard(unittest.TestCase):
+    def test_get_scorecard(self):
+        platform = "github.com"
+        org = "nexB"
+        repo = "scancode-toolkit"
+        data = get_scorecard(platform, org, repo)
+        self.assertIn("scorecard", data)
+
+if __name__ == "__main__":
```
Comment (AyanSinhaMahapatra) [inline tests/test_scorecard.py:13]: We do not run test file directly, we should us pytest.
These tests should also be added and run on the CI here so we can see it passes.

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
@404-geek Thanks!
See comments for changes requested.

File: tests/test_scorecard.py (hunk 9 above)
Comment (404-geek) [reply tests/test_scorecard.py:13]: this is also resolved

File: src/ossf_scorecard/__init__.py (hunk 8 above)
Comment (404-geek) [reply src/ossf_scorecard/__init__.py:1]: done

File: src/ossf_scorecard/scorecard_cli.py (hunk 7 above)
Comment (404-geek) [reply src/ossf_scorecard/scorecard_cli.py:25]: have removed cli code for now, will use it when it is fully ready

File: src/ossf_scorecard/scorecard_cli.py (hunk 6 above)
Comment (404-geek) [reply src/ossf_scorecard/scorecard_cli.py:17]: removed

File: src/ossf_scorecard/scorecard.py (hunk 5 above)
Comment (404-geek) [reply src/ossf_scorecard/scorecard.py:15]: done

File: src/ossf_scorecard/scorecard.py (hunk 4 above)
Comment (404-geek) [reply src/ossf_scorecard/scorecard.py:8]: have given other status code checks also 

https://github.com/nexB/ScoreCode/blob/60a880f7182feb32843ba0b964c05987edd5eebe/src/ossf_scorecard/scorecard.py#L26

File: README.rst (hunk 1 above)
Comment (404-geek) [reply README.rst:1]: https://github.com/nexB/ScoreCode/blob/60a880f7182feb32843ba0b964c05987edd5eebe/README.rst

done

File: src/ossf_scorecard/scorecard_cli.py (hunk 2 above)
Comment (404-geek) [reply src/ossf_scorecard/scorecard_cli.py:23]: removed it for now

File: src/ossf_scorecard/scorecard.py (hunk 3 above)
Comment (404-geek) [reply src/ossf_scorecard/scorecard.py:4]: yes I am checking this here

https://github.com/nexB/ScoreCode/blob/60a880f7182feb32843ba0b964c05987edd5eebe/src/ossf_scorecard/scorecard.py#L126

Comment (AyanSinhaMahapatra) [reply src/ossf_scorecard/scorecard.py:4]: Maybe we should also check if the URL exists, as an option or by default?

Comment (404-geek) [reply src/ossf_scorecard/scorecard.py:4]: Done 

https://github.com/nexB/ScoreCode/pull/5/commits/414e56d7e37dcc22013c3549e99ca3f12cc24291

**Pull request merged**